### PR TITLE
MAt-6744 added check user can delete function

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
## MADiE PR

Jira Ticket: [MAT-6744](https://jira.cms.gov/browse/MAT-6744)
(Optional) Related Tickets:

### Summary
added check user can delete function

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/util/useCheckCanDelete.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import useOktaTokens from "../hooks/useOktaTokens";
+
+const useCheckUserCanDelete = (
+  createdBy: string,
```
Comment (adongare) [inline src/util/useCheckCanDelete.ts]: I think this could be user or userName.g.
```suggestion
  user: string,
```

File: src/util/useCheckCanDelete.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  draft: boolean = true
+): boolean => {
+  const { getUserName } = useOktaTokens();
```
Comment (adongare) [inline src/util/useCheckCanDelete.ts:7]: `const { loggedInUser } = useOktaTokens();`

File: src/util/useCheckCanDelete.ts (hunk 1 above)
Comment (sb-prateekkeerthi) [reply src/util/useCheckCanDelete.ts]: thank you, I'll make the changes

File: src/util/useCheckCanDelete.ts (hunk 2 above)
Comment (sb-prateekkeerthi) [reply src/util/useCheckCanDelete.ts:7]: Ashok,  these are the only function that are present in the useOktaToken I cannot change the name to loggedInUser

useOktaTokens(storageKey?: string): {
    getAccessToken: () => any;
    getAccessTokenObj: () => any;
    getUserName: () => any;
    getIdToken: () => any;
    getIdTokenObj: () => any;
}

Comment (adongare) [reply src/util/useCheckCanDelete.ts:7]: my bad. yeah, you are right. I think I meant to change it next line. `const loggedInUser = getUserName();`

Comment (sb-prateekkeerthi) [reply src/util/useCheckCanDelete.ts:7]: Got it, thanks I'll change it

**Review by adongare — APPROVED** [review]
